apps/wakunode2/app.nim

```diff
@@ -512,6 +512,46 @@ proc setupProtocols(node: WakuNode, conf: WakuNodeConf,
       pubsubTopics = conf.topicsDeprecated.split(" ")
     else:
       pubsubTopics = conf.topics
+
+    # Validate topics and update a node's ENR accordingly
+    let parsed_topics = pubsubTopics.mapIt(NsPubsubTopic.parse(it))
+
+    var prevTopic: NsPubsubTopic
+    for i, topic in parsed_topics:
+      if topic.isErr():
+        return err("Pubsub topic parsing error")
+      
+      if i == 0:
+        prevTopic = topic.value
+        continue
+      
+      if prevTopic.kind != topic.value.kind:
+        return err("Please use named topics OR sharded ones not both.")
+        
+      if prevTopic.kind == NsPubsubTopicKind.StaticSharding:
+        if prevTopic.shard != topic.value.shard:
+          return err("Please use sharded topics within the same shard.")
+
+      if i > 0:
+        prevTopic = topic.value
+
+    if prevTopic.kind == NsPubsubTopicKind.StaticSharding:
+      let indices = parsed_topics.mapIt(it.value.shard)
+      let relayShard = RelayShards.init(prevTopic.cluster, indices)
+    
+      var record = node.wakuDiscv5.protocol.localNode.record
+      let key = node.wakuDiscv5.protocol.privKey()
+      
+      let pairs = if (let idxList = relayShard.toIndicesList(); idxList.isOk):
+        [toFieldPair(ShardingIndicesListEnrField, idxList.value)]
+      else:
+        let bitVector = relayShard.toBitVector()
+
+        [toFieldPair(ShardingBitVectorEnrField, bitVector)]
+
+      if record.update(key, pairs).isErr():
```

Comment:
wondering if there is any other higher level function we can use instead¿

not sure if `withWakuRelayShardingIndicesList` or `withWakuRelaySharding` are meant for this?

perhaps these tests can serve as an example
https://github.com/waku-org/nwaku/blob/0111b88b9b3c13065de6f09844f03c8c0259fcdc/tests/v2/test_waku_enr.nim#L337

Reply:
Now that I'm changing where discv5 will sit. I will use the EnrBuilder yes.

Reply:
Understand there may be some complexity if we don't do the code reorg now, but I'd still prefer (temporarily) extending the constructor of `wakuDiscv5` than exposing the underlying functions.